task-sdk/src/airflow/sdk/execution_time/context.py

```diff
@@ -137,10 +138,11 @@ def _get_connection(conn_id: str) -> Connection:
                 type(secrets_backend).__name__,
             )
 
-    log.debug(
-        "Connection not found in any of the configured Secrets Backends. Trying to retrieve from API server",
-        conn_id=conn_id,
-    )
+    if len(backends):
```

Comment:
```suggestion
    if backends:
```

Truthy check is fine.